Gems/Multiplayer/Code/CMakeLists.txt

```diff
@@ -9,10 +9,9 @@
 o3de_pal_dir(pal_source_dir ${CMAKE_CURRENT_LIST_DIR}/Source/Platform/${PAL_PLATFORM_NAME} "${gem_restricted_path}" "${gem_path}" "${gem_parent_relative_path}")
 
 ly_add_target(
-    NAME Multiplayer.Static STATIC
+    NAME Multiplayer.AutoGen.Static STATIC
```

Comment:
Static seems like the wrong thing to use? This should be a headers only or something similar most likely?

Reply:
This is the exact type of thing I'd love the cmake maintainers to weigh in on. It's true this target shouldn't need to be built standalone but it's also not purely headers due to AutoGen so I'm not entirely sure.

Reply:
Well an interface targer can hide the a private static target.

Such as having an `Multiplayer.AutoGen.Private.Object STATIC` static target and then having an INTERFACE target that depends on it as `Multiplayer.AutoGen.API INTERFACE`.

Arguably it would be good to split the AutoGen headers and source files up into separate directories or if that is too much work, separate `*_files.cmake` files that contain only public AutoGen API headers` and, so that any the `Multiplayer.AutoGen.API` target can be header only and doesn't have to depend on the static library.